src/features/autothread.ts

```diff
@@ -55,6 +70,16 @@ const autoThread: ChannelHandlers = {
     }
 
     const { channel: thread, author, guild } = await reaction.message.fetch();
+
+    const threadMessages = fetchMessagesByUser(
+      thread as ThreadChannel,
+      reactibot,
+    );
+
+    if (checkIfBotReplied(threadMessages)) {
```

Comment:
A couple notes:

* `fetchMessagesByUser` is a relatively expensive call, I think we should move it further down in the logic so it's only run when absolutely necessary -- we have less expensive calls that can bail out earlier
* I'd prefer to avoid `as`, as that bails out of type checking. There's a `thread.isThread()` assertion that perserves type safety. There's [an implied "return early if it's not a thread" here](https://github.com/reactiflux/reactibot/pull/180/files#diff-bdfc54759a662d98c57b52fe86e537b19ea177e8b8e2488a9a33723d2b74297bR83-R89), maybe we could make that more explicit and move this code further down to remove the need to cast